renku/core/util/git.py

```diff
@@ -823,17 +823,27 @@ def clone(branch, depth):
     try:
         # NOTE: Try to clone, assuming checkout_revision is a branch or a tag (if it is set)
         repository = clone(branch=checkout_revision, depth=depth)
-    except errors.GitCommandError:
+    except errors.GitCommandError as err:
         if not checkout_revision:
-            handle_git_exception()
-            raise
+            if raise_git_except:
+                raise
+            raise error_from_progress(progress, url)
 
         # NOTE: clone without branch set, in case checkout_revision was not a branch or a tag but a commit
         try:
             repository = clone(branch=None, depth=None)
         except errors.GitCommandError:
-            handle_git_exception()
-            raise
+            if re.match("^.* destination path .* already exists and is not an empty directory", err.stderr):
```

Comment:
Should we just unconditionally clean the directory before each clone attempt instead of the complicated error handling mechanism here?